snitun/server/worker.py

```diff
@@ -120,6 +184,11 @@ def run(self) -> None:
 
         # Shutdown worker
         _LOGGER.info("Stoping worker: %s", self.name)
+
+        # Cancel metrics task if running
+        if self._metrics_task and not self._metrics_task.done():
+            self._metrics_task.cancel()
```

Comment:
Should you also await the task? Or does it not matter that it can take some time for it to be canceled?